lib/semantics/expression.cc

```diff
@@ -1699,10 +1785,9 @@ std::optional<ActualArguments> ExpressionAnalyzer::AnalyzeArguments(
 static bool IsExternalCalledImplicitly(
     parser::CharBlock callSite, const ProcedureDesignator &proc) {
   if (const auto *symbol{proc.GetSymbol()}) {
-    return !callSite.empty() && symbol->has<semantics::SubprogramDetails>() &&
-        (symbol->owner().IsGlobal() ||
-            (symbol->owner().parent().IsGlobal() &&
-                !symbol->owner().sourceRange().Contains(callSite)));
+    return symbol->has<semantics::SubprogramDetails>() &&
+        symbol->owner().IsGlobal() &&
+        !symbol->scope()->sourceRange().Contains(callSite);
```

Comment:
I'm not positive this condition is correct. This change fixes some incorrect warnings I was getting when a generic resolved to a specific in another module. `symbol->owner()` was the module and it did not contain the call site.